src/bootstrap/src/core/config/config.rs

```diff
@@ -1468,26 +1468,35 @@ impl Config {
 
         config.stage0_metadata = build_helper::stage0_parser::parse_stage0_file();
 
-        // Find configuration file, with the following cascading fallback (first match wins):
-        // - `--config <path>`
-        // - `RUST_BOOTSTRAP_CONFIG`
-        // - `./config.toml`
-        // - `config.toml` in the root directory.
+        // Locate the configuration file using the following priority (first match wins):
+        // 1. `--config <path>` (explicit flag)
+        // 2. `RUST_BOOTSTRAP_CONFIG` environment variable
+        // 3. `./bootstrap.toml` (local file)
+        // 4. `<root>/bootstrap.toml`
+        // 5. `./config.toml` (fallback for backward compatibility)
+        // 6. `<root>/config.toml`
         let toml_path = flags
             .config
             .clone()
             .or_else(|| env::var_os("RUST_BOOTSTRAP_CONFIG").map(PathBuf::from));
         let using_default_path = toml_path.is_none();
-        let mut toml_path = toml_path.unwrap_or_else(|| PathBuf::from("config.toml"));
+        let mut toml_path = toml_path.unwrap_or_else(|| PathBuf::from("bootstrap.toml"));
+
         if using_default_path && !toml_path.exists() {
-            toml_path = config.src.join(toml_path);
+            toml_path = config.src.join(PathBuf::from("bootstrap.toml"));
+            if !toml_path.exists() {
+                toml_path = PathBuf::from("config.toml");
+                if !toml_path.exists() {
+                    toml_path = config.src.join(PathBuf::from("config.toml"));
+                }
+            }
```

Comment:
We shouldn't drop support for `config.toml` immediately. If `bootstrap.toml` is not present, we should use `config.toml` as a fallback plan.